git/file_ctime_dragonfly.go

```diff
@@ -0,0 +1,15 @@
+package git
```

Comment:
Ouch, I see what you mean. I see that you have implemented dragonfly here. The CI build runs on Linux and there are other OSes that the project aims to support. Do these other OSes need this CTime() implementation too in the near term?

Reply:
I implemented DragonFly because I wrote it on DragonFly and didn't want to spend a lot of time giving Linux special treatment since it's mostly just for the test case.

It seems to be failing under Travis (as expected):

```
not ok 70 - validate git diff-files output for a know cache/work tree state
#	
#		cat >expected <<\EOF &&
#	:100644 100644 f87290f8eb2cbbea7857214459a0739927eab154 0000000000000000000000000000000000000000 M	path0
#	:120000 120000 15a98433ae33114b085f3eb3bb03b832b3180a01 0000000000000000000000000000000000000000 M	path0sym
#	:100644 100644 3feff949ed00a62d9f7af97c15cd8a30595e7ac7 0000000000000000000000000000000000000000 M	path2/file2
#	:120000 120000 d8ce161addc5173867a3c3c730924388daedbc38 0000000000000000000000000000000000000000 M	path2/file2sym
#	:100644 100644 0aa34cae68d0878578ad119c86ca2b5ed5b28376 0000000000000000000000000000000000000000 M	path3/file3
#	:120000 120000 8599103969b43aff7e430efea79ca4636466794f 0000000000000000000000000000000000000000 M	path3/file3sym
#	:100644 100644 00fb5908cb97c2564a9783c0c64087333b3b464f 0000000000000000000000000000000000000000 M	path3/subp3/file3
#	:120000 120000 6649a1ebe9e9f1c553b66f5a6e74136a07ccc57c 0000000000000000000000000000000000000000 M	path3/subp3/file3sym
#	EOF
#		git diff-files >current &&
#		test_cmp current expected
#	
ok 71 - git update-index --refresh should succeed
not ok 72 - no diff after checkout and git update-index --refresh
```

I didn't want to add a hack for every OS I'm not using because I'm on the fence about if that's a good idea, or if we should find some other library that exposes ctime more uniformly (but then we're at their mercy as to which OSes are supported).

Reply:
Yes, the OS requirements are pretty unique for this project, so it is probably best to control this at the cost of some platform-specific code. I say this without seeing how clean/ugly the code is for the other platforms. :)

Reply:
Most of them would be pretty similar.  The stat struct would either have a different name,  or slightly different field names, or be in a different order,  but not significantly different.